command.go

```diff
@@ -1092,6 +1096,19 @@ func (c *Command) ValidateRequiredFlags() error {
 	return nil
 }
 
+// checkCommandGroups checks if a command has been added to a group that does not exists.
+// If so, we panic because it indicates a coding error that should be corrected.
+func (c *Command) checkCommandGroups() {
+	for _, sub := range c.commands {
+		// if Group is not defined let the developer know right away
+		if sub.GroupID != "" && !c.ContainsGroup(sub.GroupID) {
+			panic(fmt.Sprintf("group id '%s' is not defined for subcommand '%s'", sub.GroupID, sub.CommandPath()))
+		}
+
+		sub.checkCommandGroups()
```

Comment:
I think this introduces a minimal greater runtime overhead than checking within AddCommand, as this does an extra recursion over the subcommand tree.
But this should be neglectable.

Reply:
Since this function is only called once, the recursion will perform the check once for every single command.
I believe this is identical to when we were making the check in `AddCommand()` which is called for each command.
Isn't it?

Reply:
The check itself is identical. We just have another recursive traversion of the subcommands tree, thats all.